File: leetcode2025/2025_07/leetcode_1957_makeFancyString.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    string makeFancyString(string s) {
        s += '.';
        int n = s.size();
        int last = 0;
        string res;
        for (int i=1; i<n; i++){
            if (s[i-1] != s[i]){
                int cur = i-last;
                res += string(min(cur, 2), s[last]);
                last = i;
            }
        }
        return res;
    }
};

int main(){
    Solution sl;
    string s = "leeetcode";
    cout << sl.makeFancyString(s) << endl;
}

File: leetcode2025/2025_07/leetcode_1957_makeFancyString.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        last = 0
        s += '.'
        n = len(s)
        res = ""
        for i in range(1, n):
            if s[i-1] != s[i]:
                cur = i-last # 计算当前重复块的长度
                res += min(cur, 2)*s[last] # 一个重复块内最多选2个
                last = i
        return res

if __name__ == "__main__":
    sl = Solution()
    s = "leeetcode"
    print(sl.makeFancyString(s))
